src/cli/commands/test/mod.rs

```diff
@@ -92,6 +94,111 @@ impl From<(String, TestStatus)> for TestResult {
 	}
 }
 
+/// structure used to store a Mocked function in a cairo test entrypoint
+/// fn_name: the name of the function to mock in the cairo file
+/// fn_args: the list of parameters to replace the mock function call
+#[derive(Debug)]
+pub struct MockEntry<'a> {
+	pub fn_name: &'a str,
+	pub fn_args: &'a str,
+}
+
+/// subtype MockedFn store all the mocked functions in a single test entrypoint
+/// in a cairo file
+type MockedFn<'a> = HashMap<&'a str, Vec<MockEntry<'a>>>;
+
+/// Regex used to distinguish the beginning of a function  in a cairo file
+/// ```ignore
+/// func test_mock_call() {
+///     let mock_ret_value = 42;
+///     let func_to_mock = get_label_location(mocked_func);
+///     %{ mock_call(func_to_mock, [mock_ret_value]) %}
+///     %{ mock_call(name_of_my_func, [42, “shortstring”, ids.myVarName, Uint256 {high: 42, low: 0x0}]) %}
+///     let x = mocked_func();
+///     assert 42 = x;
+///     return ();
+///     }
+/// ```
+const FUNC_RX: &str = r"func\s+(?P<func_test_name>[\w]*)";
+
+/// Regex used to distinguish the beginning of a mock call in a cairo file
+/// ```ignore
+/// %{ mock_call(func_to_mock, [mock_ret_value]) %}
+/// ```
+const MOCK_RX: &str = r"%\{\s+mock_call\((?P<func_to_mock>.*),\s*(?P<mock_value>\[.*\])\)\s+%\}";
+
+/// compute the list of mocked functions in a text string
+/// Return a Result<MockedFn>
+///
+/// ```
+/// use cairo_foundry::cli::commands::test::extract_fname_mock_values;
```

Comment:
https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example

You can hide this import